Analysis/Tasks/Skimming_Tutorials/spectraTPC_Provider.cxx

```diff
@@ -0,0 +1,58 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+/// \author Nima Zardoshti <nima.zardoshti@cern.ch>, CERN
+
+// O2 includes
+#include "ReconstructionDataFormats/Track.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "AnalysisDataModel/PID/PIDResponse.h"
+#include "AnalysisDataModel/TrackSelectionTables.h"
+#include "AnalysisDataModel/LFDerived.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+#include "Framework/runDataProcessing.h"
+
+struct TPCSpectraProviderTask {
+
+  //Produces<aod::LFCollisions> outputCollisions; //currently it seems in the spectraTPC task no loop over the collision is made. Leave this here in case it will be added
+  Produces<aod::LFTracks> outputTracks;
+
+  //Configurable<float> cfgCutVertex{"cfgCutVertex", 10.0f, "Accepted z-vertex range"};
+  Configurable<float> trackEtaCut{"trackEtaCut", 0.9f, "Eta range for tracks"};
+
+  //Filter collisionFilter = nabs(aod::collision::posZ) < cfgCutVertex;
+  Filter trackFilter = (nabs(aod::track::eta) < trackEtaCut) && (aod::track::isGlobalTrack == (uint8_t) true);
+
+  Configurable<float> nsigmacut{"nsigmacut", 3, "Value of the Nsigma cut"}; //can we add an upper limit?
+
+  using TrackCandidates = soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::pidRespTPC, aod::TrackSelection>>;
+  void process(TrackCandidates::iterator const& track)
+  {
+    float nsigma[9] = {std::roundf(track.tpcNSigmaEl() * 100.0f) / 100.0f, std::roundf(track.tpcNSigmaMu() * 100.0f) / 100.0f,
```

Comment:
I think this is not enough to achieve on-disk compression. Have a look here this has to be done:
https://github.com/alisw/AliRoot/blob/master/STAT/AliMathBase.h#L65
https://github.com/alisw/AliPhysics/blob/master/RUN3/AliAnalysisTaskAO2Dconverter.cxx#L954

Also to avoid unneeded copying, put these right in the argument of outputTracks avoiding the additional nsigma array